src/protocols/aeternity/utils.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  AE_AMOUNT_FORMATS,
+  Encoded,
+  Tag,
+  decode,
+  formatAmount,
+} from '@aeternity/aepp-sdk';
+import BigNumber from 'bignumber.js';
+
+import {
+  IDexContracts,
+  ITransaction,
+  ITx,
+  TxFunction,
+  TxFunctionRaw,
+  TxType,
+} from '@/types';
+import {
+  AENS_NAME_MAX_LENGTH,
+  AETERNITY_CONTRACT_ID,
+  TX_FUNCTIONS,
+  TX_FUNCTIONS_TYPE_DEX,
+} from '@/popup/utils';
+import { compareCaseInsensitive, includes } from '@/utils';
+
+export function aeToAettos(value: number | string) {
+  return formatAmount(value.toString(), {
+    denomination: AE_AMOUNT_FORMATS.AE,
+    targetDenomination: AE_AMOUNT_FORMATS.AETTOS,
+  });
+}
+
+export function aettosToAe(value: number | string) {
+  return formatAmount(value.toString(), {
+    denomination: AE_AMOUNT_FORMATS.AETTOS,
+    targetDenomination: AE_AMOUNT_FORMATS.AE,
+  });
+}
+
+export function categorizeContractCallTxObject(transaction: ITransaction): {
+  amount?: string | number
+  to?: string
+  token?: string
+  url?: string
+  note?: string
+} | null {
+  if (!compareCaseInsensitive(transaction.tx.type, Tag[Tag.ContractCallTx])) {
+    return null;
+  }
+  if (transaction.incomplete || transaction.pending) {
+    const { tx } = transaction;
+    return {
+      amount: tx.amount,
+      token: tx.selectedTokenContractId ?? tx.contractId,
+      to: transaction.incomplete ? tx.recipientId : tx.callerId,
+    };
+  }
+  const { tx } = transaction as ITransaction;
+  switch (tx.function) {
+    case 'transfer':
+    case 'transfer_payload':
+    case 'change_allowance':
+    case 'create_allowance':
+      return {
+        to: tx.arguments[0].value,
+        amount: tx.arguments[1].value,
+        token: tx.contractId,
+      };
+    case 'tip_token':
+      return {
+        url: tx.arguments[0].value,
+        note: tx.arguments[1].value,
+        amount: tx.arguments[3].value,
+        token: tx.arguments[2].value,
+      };
+    case 'retip_token':
+      return {
+        url: tx.arguments[0].value,
+        amount: tx.arguments[2].value,
+        token: tx.arguments[1].value,
+      };
+    default:
+      return null;
+  }
+}
+
+/**
+ * Converts long raw values like '3280000000000000000' to human-readable 3.28
+ */
+export function getAeFee(value: number | string) {
+  return +aettosToAe(new BigNumber(value || 0).toNumber());
+}
+
+export function getTransactionPayload(transaction: ITransaction) {
+  return (transaction.tx?.payload)
+    ? decode(transaction.tx?.payload).toString()
+    : null;
+}
+
+export function getTransactionTipUrl(transaction: ITransaction): string {
+  return (
+    transaction.tipUrl
+    || transaction.url
+    || (
+      !transaction.pending
+      && !transaction.claim
+      && transaction.tx.log?.[0]
+      && transaction.tx?.function
```

Comment:
I think you should check the presence of `tx` in line 107 - there's no need of doing it in 108

Reply:
`tx` property is required in the `ITransaction` interface. So there's no need to check for it's existance.